### PR TITLE
Add SCIM role type

### PR DESCRIPTION
**Pull request by mknowlton89**
### Features and Changes

This PR introduces a new `scim` role. This role only has the ability to `manageTeam` and is meant to be used as a role type for an API key that is used to enable SCIM Provisioning. This reduces any potential for an API Key to be used for nefarious purposes.

### Testing

- [x] Confirm the new role type is visible when creating an Api Key in the app
- [x] Confirm the new role type is only visible for orgs that have access to SCIM
- [x] Confirm that this new role ISN'T an option when adding a new user or editing an existing user's global or project role.
- [x] Confirm that this new role ISN'T an option when adding a new team or editing an existing team's global or project role.
- [x] Confirm that this new role type works successfully with SCIM Provisioning (confirm you can add/remove users with this key).
- [x] Confirm that this role doesn't change the userPermissions granted for an `Admin` secret key
- [x] Confirm that this role doesn't change the userPermissions granted for a `readonly` secret key
- [x] Confirm that this role's userPermissions only includes `manageTeam`.

### Screenshots
Screen Shot 2023-12-11 at 9 34 26 AM

Screen Shot 2023-10-16 at 10 19 44 AM

### Notes

This initial implementation was done in a manner that resulted in the least amount of code changes. I added a new role type, and then I updated the `useUser` hook so that when we get `roles` for use on the InviteModal component, we filter out that new role, so an org doesn't set a user with a role of `scim`.

There is another pattern to solve this that I explored, but felt like it might be a bit too heavy handed - that is adding a new property to the `Role` type called `types`.  This types would take in an array of types. Initially it would be `user` or `api`. These types would indicate what type of resource this role can be assigned to.

And then, when could update the `useUser` hook to get userRoles and apiRoles to control which roles we use in the dropdown for the InviteModal and the ApiKeyModal.

Open to feedback if you think the initial implementation isn't great, or if you have a third implementation option.

Example permissions object for a Secret API Key with a `scim` role:
```ts
{
   createPresentations: false,
   createDimensions: false,
   createSegments: false,
   organizationSettings: false,
   superDelete: false,
   manageTeam: true, // Only permission allowed
   manageTags: false,
   manageApiKeys: false,
   manageIntegrations: false,
   manageWebhooks: false,
   manageBilling: false,
   manageNorthStarMetric: false,
   manageTargetingAttributes: false,
   manageNamespaces: false,
   manageSavedGroups: false,
   manageArchetype: false,
   viewEvents: false,
   addComments: false,
   createFeatureDrafts: false,
   manageFeatures: false,
   manageProjects: false,
   createAnalyses: false,
   createIdeas: false,
   createMetrics: false,
   manageFactTables: false,
   createDatasources: false,
   editDatasourceSettings: false,
   runQueries: false,
   publishFeatures: false,
   manageEnvironments: false,
   runExperiments: false
 }
```

### CODE REVIEW

**Comment by bttf** [comment]
Just at a glance, it seems like there are a few methods on the org controller that need to guard against `'scim'` role, but I'm not totally sure.

- `putMember`
- `putMemberRole`
- `putInviteRole`
- `addOrphanedUser`

**Comment by mknowlton89** [comment]
> Just at a glance, it seems like there are a few methods on the org controller that need to guard against `'scim'` role, but I'm not totally sure.
> 
> * `putMember`
> * `putMemberRole`
> * `putInviteRole`
> * `addOrphanedUser`

@bttf Hmm... can you expand what you mean by guarding against the `scim` role? The `scim` role only has the `manageTeam` permission. 

It does look like we might need to add a `req.checkPermissions('manageTeam')` to `putMember` - is that what you mean?

**Comment by bttf** [comment]
We [added a check in `postInvite` ](https://github.com/growthbook/growthbook/pull/1752/files#diff-b1b6cbbfd81b0b05a9ee4190596cfd51419923adc2dee43bfee1ecf24db199e5R961-R963)to prevent users from inviting a user with role set to `'scim'`. I just wonder if we need to do the same in the methods listed above?

**Comment by mknowlton89** [comment]
> We [added a check in `postInvite` ](https://github.com/growthbook/growthbook/pull/1752/files#diff-b1b6cbbfd81b0b05a9ee4190596cfd51419923adc2dee43bfee1ecf24db199e5R961-R963)to prevent users from inviting a user with role set to `'scim'`. I just wonder if we need to do the same in the methods listed above?

Ah - I see. Great catch. I've added these protections in the places listed above, as well as a few others. Thank you!

File: packages/front-end/components/Settings/ApiKeysModal.tsx [hunk 1]
```diff
@@ -117,16 +141,22 @@ const ApiKeysModal: FC<{
...
-            {
-              label: "Admin",
-              value: "admin",
-            },
-            {
-              label: "Read-only",
-              value: "readonly",
-            },
-          ]}
+          options={roleOptions.map((r) => ({
```
Comment (mknowlton89) [inline packages/front-end/components/Settings/ApiKeysModal.tsx:144]: This change improves the select dropdown options to include the role description like how we do in the invite user modal. It adds context about what each role can do.

**Comment by mknowlton89** [comment]
@bttf No rush on this, but I've made a few updates to improve the PR and make it more water tight and its ready for another review.

**Pull request closed without merging**